### PR TITLE
Add version based detection for litespeed cache plugin CVE-2024-50550

### PR DESCRIPTION
**Pull request by ybadaoui-ostorlab**
This PR is for adding a version based detection for CVE-2024-50550 which is a litespeed privilege escalation vulnerability.
this detection is based on the webexploit base class so it's only working by passing a range of the vulnerable versions which is all the versions before 6.5.2.
image

### CODE REVIEW

**Review by nmasdoufi-ol — COMMENTED** [review]
Please add a vulnerable target that was detected.

**Review by mohsinenar — APPROVED** [review]

File: agent/exploits/cve_2024_50550.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+RISK_RATING = "HIGH"
+
+
+@exploits_registry.register
+class CVE202450550Exploit(webexploit.WebExploit):
+    accept_request = definitions.Request(
+        method="GET", path="/wp-content/plugins/litespeed-cache/readme.txt"
+    )
+    check_request = definitions.Request(
+        method="GET", path="/wp-content/plugins/litespeed-cache/readme.txt"
```
Comment (oussamaessaji) [inline agent/exploits/cve_2024_50550.py:28]: Path can be const!!

**Review by ohachimOs — APPROVED** [review]

**Review by ybadaoui-ostorlab — COMMENTED** [review]
@nmasdoufi-ol please check again i did update the description with a real target finding screenshot

**Review by rabsondev — APPROVED** [review]

**Review by PiranhaSa — APPROVED** [review]
LGTM

**Review by elyousfi5 — APPROVED** [review]

**Review by adnaneserrar — COMMENTED** [review]
LGTM

File: agent/exploits/cve_2024_50550.py (hunk 1 above)
Comment (benyissa) [reply agent/exploits/cve_2024_50550.py:28]: The occurrences of this path is 5 times. I would put it into a const too.

Comment (ybadaoui-ostorlab) [reply agent/exploits/cve_2024_50550.py:28]: it's not always the same path for both accept and check so it's more clear this way!

**Review by nmasdoufi-ol — APPROVED** [review]

**Review by 3asm — APPROVED** [review]

**Pull request merged**